kotlinx-coroutines-core/build.gradle

```diff
@@ -205,12 +206,26 @@ task jvmStressTest(type: Test, dependsOn: compileTestKotlinJvm) {
     systemProperty 'kotlinx.coroutines.semaphore.maxSpinCycles', '10'
 }
 
+task jvmLincheckTest(type: Test, dependsOn: compileTestKotlinJvm) {
+    classpath = files { jvmTest.classpath }
+    testClassesDirs = files { jvmTest.testClassesDirs }
+    minHeapSize = '1g'
+    maxHeapSize = '10g' // we may need more space for building an interleaving tree in the model checking mode
```

Comment:
Min memory requirements for our teamcity agents are currently set to 6Gb. We need to investigate if it can be raised to > 10Gb without the risk of losing agents in the pool.

Reply:
6Gb is enough, change the `maxHeapSize` correspondingly and checked on [a Linux nightly build](https://teamcity.jetbrains.com/buildConfiguration/KotlinTools_KotlinxCoroutines_NightlyStress/3180685?buildTab=tests).